component/loki/write/write.go

```diff
@@ -159,6 +160,13 @@ func (c *Component) Update(args component.Arguments) error {
 	}
 
 	cfgs := newArgs.convertClientConfigs()
+	uid := agentseed.Get().UID
+	for _, cfg := range cfgs {
+		if cfg.Headers == nil {
+			cfg.Headers = map[string]string{}
+		}
+		cfg.Headers["X-Agent-UID"] = uid
```

Comment:
can we define the header name somewhere as a constant? (seems to be repeated in multiple places)